huaweicloud/services/ga/resource_huaweicloud_ga_accelerator.go

```diff
@@ -485,9 +487,87 @@ func resourceAcceleratorUpdate(ctx context.Context, d *schema.ResourceData, meta
 			return diag.Errorf("error waiting for the Update of Accelerator (%s) to complete: %s", d.Id(), err)
 		}
 	}
+
+	// update tags
+	if d.HasChange("tags") {
+		client, err := conf.NewServiceClient("ga", region)
+		if err != nil {
+			return diag.Errorf("error creating GA Client: %s", err)
+		}
+
+		oldRaw, newRaw := d.GetChange("tags")
+		oldMap := oldRaw.(map[string]interface{})
+		newMap := newRaw.(map[string]interface{})
+
+		// remove old tags
+		if len(oldMap) > 0 {
+			deleteTagsHttpUrl := "v1/{resource_type}/{resource_id}/tags/delete"
+			if err = deleteTags(client, deleteTagsHttpUrl, "ga-accelerators", d.Id(), oldMap); err != nil {
+				return diag.FromErr(err)
+			}
+		}
+
+		// set new tags
+		if len(newMap) > 0 {
+			createTagsHttpUrl := "v1/{resource_type}/{resource_id}/tags/create"
+			if err := createTags(client, createTagsHttpUrl, "ga-accelerators", d.Id(), newMap); err != nil {
+				return diag.FromErr(err)
+			}
+		}
+	}
+
 	return resourceAcceleratorRead(ctx, d, meta)
 }
 
+func createTags(createTagsClient *golangsdk.ServiceClient, createTagsHttpUrl, resourceType, resourceId string,
+	tags map[string]interface{}) error {
+	if len(tags) > 0 {
```

Comment:
This step is not necessary, you have already made a judgment before calling this method